lib/travis/api/v3/service.rb

```diff
@@ -89,21 +89,23 @@ def result_type
       self.class.result_type
     end
 
-    def result(*args)
-      Result.new(access_control, *args)
+    def result(resource, **meta_data)
+      return not_found unless resource
+      rt = meta_data[:result_type] || result_type
+      status = meta_data[:status] || 200
```

Comment:
I might be reading it wrongly, but is this going to pass `result_type` and `status` to the constructor twice?